util/keyfile/src/bin/main.rs

```diff
@@ -5,18 +5,38 @@
 //! Reads .bin file on stdin, or a path to .bin file, emits description on
 //! stdout
 
-use mc_account_keys::{AccountKey, RootIdentity};
+use mc_account_keys::AccountKey;
+use std::{
+    env, fs, io,
+    io::{Cursor, Read},
+};
 
-fn main() {
-    let root_id: RootIdentity = {
-        let args: Vec<String> = std::env::args().collect();
-        match args.get(1) {
-            None => mc_util_keyfile::read_keyfile_data(&mut std::io::stdin())
-                .unwrap_or_else(|_| panic!("Failed when reading from stdin")),
-            Some(arg) => mc_util_keyfile::read_keyfile(arg)
-                .unwrap_or_else(|_| panic!("Failed when reading from {}", arg)),
-        }
+fn print_keyfile_bytes(bytes: Vec<u8>) {
+    let acct_key = if let Ok(identity) =
+        mc_util_keyfile::read_root_entropy_keyfile_data(Cursor::new(bytes.as_slice()))
+    {
+        println!("{:?}", identity);
+        AccountKey::from(&identity)
+    } else {
+        mc_util_keyfile::read_keyfile_data(Cursor::new(bytes.as_slice()))
+            .expect("Could not parse key file as either mnemonic or legacy entropy")
     };
-    let acct_key = AccountKey::from(&root_id);
-    println!("{:?}\n{:?}", root_id, acct_key,);
+
+    println!("{:?}", acct_key);
+}
+
+fn main() {
+    let mut n_files = 0usize;
+    for path in env::args().skip(1) {
+        print_keyfile_bytes(fs::read(path).expect("Could not read file"));
```

Comment:
Let's include the path
```suggestion
        print_keyfile_bytes(fs::read(path).expect(&format!("Could not read file '{}'", path)));
```